be/src/vec/exec/format/parquet/vparquet_reader.cpp

```diff
@@ -180,8 +215,19 @@ Status ParquetReader::init_reader(
         return Status::EndOfFile("Empty Parquet File");
     }
     auto schema_desc = _file_metadata->schema();
+    if (_schema_evolution) {
+        _gen_col_name_maps(_col_id_name_map);
+    }
     for (int i = 0; i < schema_desc.size(); ++i) {
-        _map_column.emplace(schema_desc.get_column(i)->name, i);
+        if (_schema_evolution) {
+            if (_file_col_to_table_col.find(schema_desc.get_column(i)->name) !=
```

Comment:
Extract the `_file_col_to_table_col.find(schema_desc.get_column(i)->name)` so that we don't need to call it twice